### PR TITLE
docs: add eslint setup guide

### PR DESCRIPTION
**Pull request by chris-visser**
I found @antfu 's eslint package to be the best approach to enable eslint in a Nuxt project and since imho its not really clear on how to set up eslint etc in Nuxt, I've documented it. So if people agree lets add it to the docs?

### CODE REVIEW

File: docs/2.guide/1.concepts/9.code-style.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  ```bash [pnpm]
+  pnpm add -D eslint @antfu/eslint-config
+  ```
+
+  ```bash [bun]
+  bun add -D eslint @antfu/eslint-config
+  ```
+
+**Add eslint.config.js to the root folder of your Nuxt app**
```
Comment (skmedix) [inline docs/2.guide/1.concepts/9.code-style.md:28]:
```suggestion
**Add `eslint.config.js` to the root folder of your Nuxt app**
```

**Comment by MuhammadM1998** [comment]
There's a WIP https://github.com/nuxt/eslint-plugin-nuxt  made by the nuxt team so I think this would be the official way.
P.S: the description says nuxt supports prettier but antfu config uses stylistic instead of prettier.

**Review by huang-julien — COMMENTED** [review]
Hello :wave:
we do have a roadmap for nuxt/eslint https://github.com/nuxt/eslint-config/issues/303
Currently i'm using `@nuxtjs/eslint-config-typescript` which works quite well i'd say

**Comment by antfu** [comment]
Hey, thanks for trying!

I am working on a Nuxt module to generate a project-aware Nuxt configuration to make this easier. As the ESLint flat config is slowly becoming the standard, I believe it would make the best DX for Nuxt project.

(while me, who made my own eslint config, do believe my eslint config works pretty well ofc. It's still quite opininated to my personal taste, and might not work for every Nuxt project. It would be better to maintain a Nuxt-official version and recommend it (which we do, but there could be a lot of improvements to make for sure)).

**Comment by chris-visser** [comment]
> There's a WIP https://github.com/nuxt/eslint-plugin-nuxt made by the nuxt team so I think this would be the official way. P.S: the description says nuxt supports prettier but antfu config uses stylistic instead of prettier.

I've noticed, but the last update on that package was done on 2022. I do agree though that an official package would be much better. What I find interesting though is the zero config setup of Antfu's package which is in line with Nuxt's approach to many things like the module system. 

I'm fine with anything though. Just trying to help :)

**Comment by MuhammadM1998** [comment]
@chris-visser I appreciate your help  🙏and I'm just trying to do the same didn't mean to be aggressive or something. FYI I'm using antfu's package in my project and it's indeed zero-config and it's the Nuxt way as you said. I hope that official plugin get stabilized soon!

**Comment by chris-visser** [comment]
> @chris-visser I appreciate your help 🙏and I'm just trying to do the same didn't mean to be aggressive or something. FYI I'm using antfu's package in my project and it's indeed zero-config and it's the Nuxt way as you said. I hope that official plugin get stabilized soon!

Appreciate it !

**Review by danielroe — CHANGES_REQUESTED** [review]
Let's update this to use `@nuxt/eslint-config` for the moment. 🙏

**Comment by Atinux** [comment]
Agree with @danielroe 

I do use `@nuxt/eslint-config` in most of my Nuxt projects and it works well ([example](https://github.com/Atinux/nuxt-todos-edge/blob/main/.eslintrc.cjs)).

Thank you for pushing this new page in the documentation @chris-visser ❤️

File: docs/2.guide/1.concepts/9.code-style.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+---
+title: 'Code Style'
+description: "Nuxt 3 supports eslint out of the box"
+---
+
+## Eslint
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md:6]:
```suggestion
## ESLint
```

File: docs/2.guide/1.concepts/9.code-style.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+The recommended approach for Nuxt is to enable eslint using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md:8]:
```suggestion
The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config). 
```

File: docs/2.guide/1.concepts/9.code-style.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+```json
+  "scripts": {
+    ...
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
+    ...
+  },
+```
+
+Run the lint command to check if the codestyle is correct or run `lint:fix` to automatically fix issues.
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md:50]:
```suggestion
Run the lint command to check if the code style is correct or run `lint:fix` to automatically fix issues.
```

File: docs/2.guide/1.concepts/9.code-style.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+---
+title: 'Code Style'
+description: "Nuxt 3 supports eslint out of the box"
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md:3]:
```suggestion
description: "Nuxt supports ESLint out of the box"
```

File: docs/2.guide/1.concepts/9.code-style.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+```json
+{
+  "editor.formatOnSave": true,
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  }
+}
+```
+
+You're good to go! On save, your files will be formatted.
```
Comment (antfu) [inline docs/2.guide/1.concepts/9.code-style.md:67]:
```suggestion
You're good to go! On file save, your files will be lint with auto-fix.
```

We should maybe mention that `@nuxt/eslint-config` does not do auto-formatting for you, and maybe guide ppl to set up Prettier?

File: docs/2.guide/1.concepts/9.code-style.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+Install the [VSCode ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint).
+
+In VSCode press `ctrl+shift+p` (`cmd+shift+p` on Mac) to open the command prompt, find `Open Workspace Settings (JSON)`, add the below lines to the JSON and save:
+
+```json
+{
+  "editor.formatOnSave": true,
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
```
Comment (antfu) [inline docs/2.guide/1.concepts/9.code-style.md:62]:
```suggestion
    "source.fixAll.eslint": "explicit"
```

It's updated in the latest VS Code

File: docs/2.guide/1.concepts/9.code-style.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+## Configuring VSCode
+
+Install the [VSCode ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint).
+
+In VSCode press `ctrl+shift+p` (`cmd+shift+p` on Mac) to open the command prompt, find `Open Workspace Settings (JSON)`, add the below lines to the JSON and save:
+
+```json
+{
+  "editor.formatOnSave": true,
```
Comment (antfu) [inline docs/2.guide/1.concepts/9.code-style.md:60]: I am not sure if enabling this has something to do with ESLint (`codeActionsOnSave` is different from `formatOnSave`, I believe). In my own usage, I set this off to disable Prettier in favor of ESLint's auto-fix. I guess for `@nuxt/eslint-config`, we don't need to mention this line as there is no formatting involved.

```suggestion
```

**Review by antfu — APPROVED** [review]
Otherwise LGTM!

And btw, as promised, here I made a project-aware flat config generator module: https://github.com/antfu/nuxt-module-eslint-config, at some point we might make it official and mention it as the experiment support.

For better UX, I am thinking maybe this module could add an optional flag to opt-in ESLint for formatting via [ESLint Stylisttc](https://eslint.style) and [`eslint-plugin-format`](https://github.com/antfu/eslint-plugin-format) (use Prettier for HTML and CSS).

File: docs/2.guide/1.concepts/9.code-style.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+```json
+  "scripts": {
+    ...
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
+    ...
+  },
+```
+
+Run the lint command to check if the code style is correct or run `lint:fix` to automatically fix issues.
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md:58]:
```suggestion
Run the `lint` command to check if the code style is correct or run `lint:fix` to automatically fix issues.
```

File: docs/2.guide/1.concepts/9.code-style.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
+---
+title: 'Code Style'
+description: "Nuxt supports ESLint out of the box"
+---
+
+## ESLint
+
+The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md]: what about this @antfu?

```suggestion
The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).

At the moment, this configuration will not format your files; you can set up Prettier or another tool to do so.

::alert{type=info}
We're currently working to refactor the Nuxt ESLint configuration. Subscribe to [the Nuxt ESLint roadmap](https://github.com/nuxt/eslint-config/issues/303) to follow updates.]
::
```

File: docs/2.guide/1.concepts/9.code-style.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
```
Comment (danielroe) [inline docs/2.guide/1.concepts/9.code-style.md:8]: what about this @antfu?

```suggestion
The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).

At the moment, this configuration will not format your files; you can set up Prettier or another tool to do so.

::alert{type=info}
We're currently working to refactor the Nuxt ESLint configuration. Subscribe to [the Nuxt ESLint roadmap](https://github.com/nuxt/eslint-config/issues/303) to follow updates.]
::
```

**Pull request merged**